src/contract.rs

```diff
@@ -80,27 +93,35 @@ pub mod execute {
             fallback: false,
         };
 
-        let mut tokens: Vec<Token> = vec![];
+        let mut tokens_id: Vec<Token> = vec![];
+        let mut tokens_take_profit: Vec<Token> = vec![];
         let retry_delay: u64 = RETRY_DELAY.load(deps.storage)?;
-        for deposit_id in deposit_ids {
+        let mut i = 0;
+        while i < deposit_ids.len() {
```

Comment:
This is not the way to do a for loop. Use
```
for deposit in deposits
```
or similar.

Reply:
(
for this _exact_ circumstance you would normally use one of the zip functions
https://doc.rust-lang.org/std/iter/fn.zip.html
or
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.zip
)

Reply:
fixed